snap/snapcraft.yaml

```diff
@@ -1,56 +1,83 @@
+description: Refer to https://snapcraft.io/matter-pi-gpio-commander
+
 grade: devel
+confinement: strict
 
+base: core22
 architectures:
   - build-on: arm64
   - build-on: armhf
+  # - build-on: amd64
 
 layout:
   /mnt:
-    bind: $SNAP_DATA/mnt
+    bind: $SNAP_COMMON/mnt
 
 parts:
-  build-res:
-    plugin: dump
-    source: build
+  wiring-pi:
+    plugin: nil
+    source: https://github.com/WiringPi/WiringPi.git
+    source-depth: 1
+    source-tag: 2.61-1
     override-build: |
-      cp requirements.txt $CRAFT_STAGE/
+      WIRINGPI_SUDO="" ./build
+
+      mkdir -p $CRAFT_PART_INSTALL/lib
+      cp -v /usr/local/lib/libwiringPi.so* $CRAFT_PART_INSTALL/lib/
+    build-packages: [git, gcc, g++]
+
+  test-blink:
+    after: [wiring-pi]
+    plugin: nil
+    source: .
+    override-build: |
+      g++ -Wall -o test-blink test-blink.cpp -lwiringPi
+      mkdir -p $CRAFT_PART_INSTALL/bin
+      cp test-blink $CRAFT_PART_INSTALL/bin/
+  
   connectedhomeip:
-    after: [build-res]
     plugin: nil
-    override-build: |   
-      git clone https://github.com/project-chip/connectedhomeip.git --depth=1 .
+    source: https://github.com/project-chip/connectedhomeip.git
+    source-depth: 1
+    source-tag: v1.0.0.2
+    source-submodules: []
+    override-pull: |
+      craftctl default
+      # shallow clone the submodules
       scripts/checkout_submodules.py --shallow --platform linux
 
-      # The project writes its data to /tmp. Snap does not allow bind mounting
-      #   a persistent directory on /tmp, so we need to replace it in the code.
-      # Replace all /tmp directories with /mnt:
+  lighting:
+    after: [wiring-pi, connectedhomeip]
+    plugin: nil
+    source: app
+    override-build: |
+      cd ../../connectedhomeip/src
+
+      # Replace storage paths:
       sed -i 's/\/tmp/\/mnt/g' src/platform/Linux/CHIPLinuxStorage.h
+      # Replace key-value store path:
       sed -i 's/\/tmp/\/mnt/g' src/platform/Linux/CHIPPlatformConfig.h
 
-      set +u # do not treat unset variables as error
-      # Setup the compilation environment
+      # To avoid unrelated activation errors, don't treat unset variables as error
+      set +u
       source scripts/activate.sh
+      set -u
+      
+      cd examples/lighting-app/linux/
 
-      # Python setup
-      ./scripts/build_python_device.sh --chip_detail_logging true
-      source out/python_env/bin/activate
-
-      # This is needed for the gpio. Install it here since the venv is active.
-      pip install -r $CRAFT_STAGE/requirements.txt
+      # Copy and replace the application files
+      cp -vr $CRAFT_PART_BUILD/* .
 
-      sed 's/^VIRTUAL_ENV=.*/VIRTUAL_ENV="\$SNAP\/connectedhomeip\/python_env"/' \
-        out/python_env/bin/activate > out/python_env/bin/activate_snap
+      gn gen out/build
+      ninja -C out/build
 
-      mkdir -p $CRAFT_PART_INSTALL/connectedhomeip
-      cp -r out/python_env $CRAFT_PART_INSTALL/connectedhomeip/
+      ldd out/build/chip-lighting-app
```

Comment:
Is this for debugging?

Reply:
This is just to see the linked objects in the build logs.